code/lib/types/src/modules/csf.ts

```diff
@@ -135,13 +138,15 @@ export type ViewMode = ViewModeBase | 'story' | 'info' | 'settings' | string | u
 
 type Layout = 'centered' | 'fullscreen' | 'padded' | 'none';
 
-export interface Parameters extends ParametersBase {
-  fileName?: string;
+export interface StorybookParameters {
   options?: Addon_OptionsParameter;
   /** The layout property defines basic styles added to the preview body where the story is rendered. If you pass 'none', no styles are applied. */
   layout?: Layout;
-  docsOnly?: boolean;
-  [key: string]: any;
+}
```

Comment:
I'm not quite sure I understand the naming here. I thought we had decided we would shadow the CSF names (e.g. we export our own `Renderer` rather than creating a `StorybookRenderer`). Did you change your mind about that? Or is there something special about `Parameters`?

Reply:
I know about that decision, and I kind of agreed on it. But the problem is that users can not use module augmentation anymore on `Parameters` if we shadow the type in `@storybook/csf`.

So if shadow the csf types in `@storybook/types` and extend of it, and maybe also do the same in a renderer. For example in `@storybook/angular` we could shadow `@storybook/types` again and enhance it:

```ts
import { Parameters as StorybookParameters } from '@storybook/types';

export interface Parameters extends StorybookParameters {
  /** Uses legacy angular rendering engine that use dynamic component */
  angularLegacyRendering?: boolean;
  bootstrapModuleOptions?: unknown;
}; 
```

Then if users tries to do module augmentation to add Chromatic or Addon specific parameters to this:

```ts
declare module '@storybook/angular' {
  interface Parameters
    extends ChromaticParameters,
      DocsParameters,
      BackgroundParameters {}
}
```

We are not augmenting the types of csf anymore. So it won't get autocompletion for parameters in all the places where CSF is used (Meta/StoryObj/StoryFn/Decorator/Loader/StoryContext etc.).

So instead I use really export the reference of the csf `Parameters` in `@storybook/csf` and again in `@storybook/angular` so that the user can augment it like this:

```ts
import type { AngularParameters, StorybookParameters } from '@storybook/angular';
import type { ChromaticParameters } from 'chromatic';
import type { DocsParameters } from '@storybook/addon-docs';
import type { BackgroundsParameters } from '@storybook/addon-backgrounds';

declare module '@storybook/angular' {
  interface Parameters
    extends StorybookParameters,
      AngularParameters,
      ChromaticParameters,
      DocsParameters,
      BackgroundsParameters {}
}
```

Reply:
Ok, this makes sense to me. Thanks for explaining it @kasperpeulen. I guess we treat `StorybookParameters` (and maybe `StorybookGlobals`, although I don't think there are any) as a special case then?

Reply:
Yes, exactly!